### PR TITLE
aws-vault 6.2.0 (new formula)

### PR DESCRIPTION
**Pull request by issyl0**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

- This is a Linux-only formula. The macOS version ships as a Cask.
- Michka asked me to do this in https://github.com/Homebrew/linuxbrew-core/pull/21271.

### CODE REVIEW

**Review by iMichka — CHANGES_REQUESTED** [review]
Looks like it needs a license.

I am wondering if this will not conflict with the cask on Mac, and if we should give the formula another name?

**Comment by SMillerDev** [comment]
It will, but it would be even nicer if it didn't have to be a cask

**Comment by issyl0** [comment]
I remember having this discussion at the time I added the Linux-only formula - something about signed binaries for keychain access?

**Comment by issyl0** [comment]
It's not necessarily a good user experience, but Homebrew at least tells you its preference for the formula while also appropriately failing to install the formula on macOS:

```
➜ brew install -s aws-vault
Warning: Treating aws-vault as a formula. For the cask, use homebrew/cask/aws-vault
aws-vault: Linux is required.
Error: An unsatisfied requirement failed this build.
```

So yeah, worth a rename I think. Or, "unsatisfied requirement" related to mismatched OSes should default to the cask?

**Comment by issyl0** [comment]
Hold on a second... how does `docker` do it? That's both a Formula and a Cask with the same name (see `brew info docker` and `brew cask info docker`.) That's worth looking into...

File: Formula/aws-vault.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  license "MIT"
+
+  depends_on "go" => :build
+  depends_on :linux
+
+  def install
+    flags = "-X main.Version=#{version} -s -w"
+
+    system "go", "build", "-ldflags=#{flags}"
+    bin.install "aws-vault"
```
Comment (dawidd6) [inline Formula/aws-vault.rb:15]:
```suggestion
```

File: Formula/aws-vault.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  sha256 "39886f4bc3985d4aefbae6fc88532499ac2c39cbabc33d860bba6d355158e17d"
+  license "MIT"
+
+  depends_on "go" => :build
+  depends_on :linux
+
+  def install
+    flags = "-X main.Version=#{version} -s -w"
+
+    system "go", "build", "-ldflags=#{flags}"
```
Comment (dawidd6) [inline Formula/aws-vault.rb:14]:
```suggestion
    system "go", "build", *std_go_args, "-ldflags=#{flags}"
```

**Comment by MikeMcQuaid** [comment]
> I remember having this discussion at the time I added the Linux-only formula - something about signed binaries for keychain access requiring it to be a Cask on macOS?

If this is still the blocker: I wonder if it will work with self-signing i.e. using `codesign -s`. If so, support for this was recently added to Ruby MachO and we could consider adding a DSL to do this for formulae in general (I believe GDB also requires this CC @alebcay).

Can someone test this out and see? CC @mistydemeo @woodruffw who have been previously talking about signing.

**Review by SMillerDev — DISMISSED** [review]
Thanks @issyl0  ! Without your contributions it'd be impossible to keep homebrew going. You can feel good knowing that you've made the world a lot better for homebrew users around the world! 👍 🎉

**Comment by dawidd6** [comment]
@SMillerDev we just squash-and-merge PRs labelled as `linux to homebrew-core`. Without bottles.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]
Can we hold off on this until we figure out the cask situation, please?

**Comment by issyl0** [comment]
> Can we hold off on this until we figure out the cask situation, please?

That's exactly why I'd not merged it myself yet. 😅 I haven't had time to test the codesigning stuff yet (and I know nothing about codesigning).

**Comment by MikeMcQuaid** [comment]
It feels like the tooling shouldn't have allowed triggering a merge when @iMichka had unresolved requested changes. CC @dawidd6 @jonchang for thoughts on that.

**Pull request closed without merging**